Userland/Libraries/LibGUI/EditingEngine.h

```diff
@@ -29,6 +29,8 @@ class EditingEngine {
     void attach(TextEditor& editor);
     void detach();
 
+    TextEditor* editor() { return m_editor.unsafe_ptr(); }
```

Comment:
you VERIFY this is non-null in all the places it is used, so making it return a reference will probably be cleaner.